sage/index.py

```diff
@@ -42,6 +53,14 @@ def main():
     if args.embedding_provider == "marqo" and args.vector_store_provider != "marqo":
         parser.error("When using the marqo embedder, the vector store type must also be marqo.")
 
+    # We need nltk tokenizers for 
+    if is_punkt_downloaded():
+        print("punkt is already downloaded")
+    else:
+        print("punkt is not downloaded")
+        # Optionally download it
+        nltk.download('punkt_tab')
```

Comment:
We only need nltk when using BM25, so I would move this somewhere [here](https://github.com/Storia-AI/sage/blob/40357dc0b8359063ffa46255c54d96038c0ccecf/sage/vector_store.py#L71) so that we don't unnecessarily download it when not needed.

Reply:
Plus you want to make sure that the download happens even if someone runs `sage-chat` without running `sage-index`.